text/0229-deprecate-testing-restricted-resolver.md

```diff
@@ -0,0 +1,217 @@
+  * `setupModelTest`
+
+### Non Component Test APIs
+
+The migration path for `moduleFor`, `moduleForModel`, `setupTest`, and 
+`setupModelTest` is very simple:
+
+```js
+// ember-qunit
+
+// before
+moduleFor('service:session');
+
+moduleFor('service:session', {
+  unit: true
+});
+
+moduleFor('service:session', {
+  needs: ['type:thing']
+});
+
+// after
+moduleFor('service:session', {
+  integration: true
+});
+```
+
+```js
+// ember-mocha
+
+// before
+describe('Session Service', function() {
+  setupTest('service:session');
+  
+  // ...snip...
+});
+
+describe('Session Service', function() {
+  setupTest('service:session', { unit: true });
+  
+  // ...snip...
+});
+
+describe('Session Service', function() {
+  setupTest('service:session', { needs: [] });
+  
+  // ...snip...
+});
+
+// after
+
+describe('Session Service', function() {
+  setupTest('service:session', { integration: true });
+  
+  // ...snip...
+});
+```
+
+The main change is adding `integration: true` to options (and removing `unit` or `needs`
+if present).
+
+### Component Test APIs
+
+Implicitly relying on "unit test mode" has been deprecated for quite some time 
+([introduced 2015-04-07](https://github.com/emberjs/ember-test-helpers/pull/38)),
+so all consumers of `moduleForComponent` and `setupComponentTest` are specifying
+one of the "test type options" (`unit`, `needs`, or `integration`).
+
+This RFC proposes to deprecate completely using `unit` or `needs` options with
+`moduleForComponent` and `setupComponentTest`. The vast majority of component tests
+should be testing via `moduleForComponent` / `setupComponentTest` with the `integration: true`
+option set, but on some rare occaisions it is easier to use the "unit test" style is
+desired (e.g. non-rendering test) these tests should be migrated to using `moduleFor` 
+/ `setupTest` directly.
+
+```js
+// ember-qunit
+
+// before
+moduleForComponent('display-page', {
+  unit: true
+});
+
+moduleForComponent('display-page', {
+  needs: ['type:thing']
+});
+
+// after
+
+moduleFor('component:display-page', {
+  integration: true
+});
+```
+
+```js
+// ember-mocha
+describe('DisplayPageComponent', function() {
+  setupComponentTest('display-page', { unit: true });
+  
+  // ...snip...
+});
+
+describe('DisplayPageComponent', function() {
+  setupComponentTest('display-page', { needs: [] });
+  
+  // ...snip...
+});
+
+// after
+
+describe('DisplayPageComponent', function() {
+  setupTest('component:display-page', { integration: true });
+  
+  // ...snip...
+});
+```
+
+## Ecosystem Updates
+
+The blueprints in all official projects (and any provided by popular
+addons) will need to be updated to avoid triggering a deprecation.
+
+This includes:
+
+* `ember-source`
+* `ember-data`
+* `ember-cli-legacy-blueprints`
+* Others?
+
+## Remove Deprecated `unit` / `needs` Options
+
+Once the changes from this RFC are made, we will be able to remove
+support for `unit` and `needs` options from `ember-test-helpers`,
```

Comment:
@trentmwillis - OK, that seems like a reasonable compromise.  Basically a single deprecation mentioning that `integration: true` has no effect and should be removed.

Also, for context, I am working on another RFC that will suggest migrating to a new ember-qunit API (see https://github.com/emberjs/ember-qunit/pull/258 for context). My plan is to land both of these RFCs in roughly the same timeframe, so that we can update the blueprints once. The new API will not allow `unit`, `needs`, or `integration` options, and will likely suggest deprecating the current style completely (which is why I am not terribly concerned with leaving `integration` around for long...).